### PR TITLE
fix(salesforce): cleanup logic

### PR DESCRIPTION
**Pull request by evan-onyx**
## Description

there were some edge cases we weren't handling in the salesforce connector; now each run is "fresh".

## How Has This Been Tested?

hard to test the specific changes, will rely on CI

## Additional Options

- [x] [Required] I have considered whether this PR needs to be cherry-picked to the latest beta branch.
- [x] [Optional] Override Linear Check

---
## Summary by cubic
Make each Salesforce sync run start fresh and harden SQLite handling to prevent stale state and disk I/O errors. Also ensure document metadata uses valid string types.

- **Bug Fixes**
  - Always use a temp directory for SQLite in full and delta syncs to avoid collisions and stale WAL/SHM files.
  - Automatically recover from SQLite "disk I/O error" by removing db/WAL/SHM files and recreating the database.
  - Convert Salesforce field values to metadata-safe types (str or list[str]) to prevent type issues.

<sup>Written for commit 35f6d0f278cd20d07c9c0281903cc3608b25a564. Summary will update on new commits.</sup>

### CODE REVIEW

File: backend/onyx/connectors/salesforce/connector.py [hunk 1]
```diff
@@ -1089,36 +1100,21 @@ def _make_context(
...
-        sqlite_db_path = get_sqlite_db_path(BASE_DATA_PATH)
-        if os.path.exists(sqlite_db_path):
-            logger.info(f"load_from_state: Removing db at {sqlite_db_path}.")
-            os.remove(sqlite_db_path)
-        return self._full_sync(BASE_DATA_PATH)
+        # Always use a temp directory for SQLite - the database is rebuilt
+        # from scratch each time via CSV downloads, so there's no caching benefit
+        # from persisting it. Using temp dirs also avoids collisions between
+        # multiple CC pairs and eliminates stale WAL/SHM file issues.
+        # TODO(evan): make this thing checkpointed and persist/load db from filestore
```
Comment (nmgarza5) [inline backend/onyx/connectors/salesforce/connector.py:1123]: linear ticket for this please if you dont plan on following up asap

File: backend/onyx/connectors/salesforce/sqlite_functions.py [hunk 2]
```diff
@@ -99,8 +103,37 @@ def flush(self) -> None:
...
-        Non-destructive operation.
+        Non-destructive operation. If a disk I/O error is encountered (often due
+        to stale WAL/SHM files from a previous crash), this method will attempt
+        to recover by removing the corrupted files and recreating the database.
         """
+        try:
+            self._apply_schema_impl()
+        except sqlite3.OperationalError as e:
+            if SQLITE_DISK_IO_ERROR not in str(e):
+                raise
```
Comment (nmgarza5) [inline backend/onyx/connectors/salesforce/sqlite_functions.py:114]: noob question...does the appropriate error bubble up here still?
